account-integrations/zkp/relayer/src/lib/imapClient.ts

```diff
@@ -4,55 +4,61 @@ type EmailResponse = {
     headers: Buffer;
     sender: string;
     subject: string;
+    uid: number;
 };
 
 class ImapClient {
-    private imapClient: ImapFlow;
+    public imapClient: ImapFlow;
 
-    constructor(imapConfig: ImapFlowOptions) {
-        this.imapClient = new ImapFlow(imapConfig);
+    constructor(imapClientConfig: ImapFlowOptions) {
+        this.imapClient = new ImapFlow(imapClientConfig);
     }
 
-    public async start(): Promise<void> {
+    public async start() {
         await this.imapClient.connect();
     }
 
-    public async stop(): Promise<void> {
+    public async stop() {
         await this.imapClient.logout();
     }
 
     public async fetchEmails(): Promise<Array<EmailResponse>> {
         const lock = await this.imapClient.getMailboxLock("INBOX");
-        const emails = new Array<EmailResponse>();
 
+        const emails = new Array<EmailResponse>();
         try {
-            const unreadMessages = this.imapClient.fetch(
-                { seen: false },
-                {
-                    headers: true,
-                    envelope: true,
-                    source: true,
-                    bodyStructure: true,
-                    flags: true,
-                }
-            );
+            // For some reason calling .status() seems to "refresh" the inbox so that
+            // new emails can be detected. Without this line, new emails are not detected.
+            const mailbox = await this.imapClient.status("INBOX", {
```

Comment:
This is the line that I added that enabled the imap client to fetch new emails. `this.imapClient.fetch` on its own did not suffice. 

Another approach I tried that worked was to persist an imapClient instance over the course of this function. E.g. `const imapClient = new ImapFlow(imapClientConfig)`, and then start and end the connection in this function. It felt better to connect to an inbox once and persist the connection, rather than connecting and logging out everytime the inbox is being polled - you can see this approach in the `Receive new emails` commit.